test/client_ipc.cpp

```diff
@@ -6,34 +6,36 @@
 * The full license is in the file LICENSE, distributed with this software. *
 ****************************************************************************/
 
-// TODO: this client should be reimplemented with
-// the new client framework. Notice that it was not
-// tested in the CI but was here for trying to reproduce
-// some issue with IPC.
-
-/*#include <iostream>
+#include <iostream>
 
 #include "zmq_addon.hpp"
 #include "nlohmann/json.hpp"
 #include "xeus/xguid.hpp"
 #include "xeus/xmessage.hpp"
-#include "xeus-zmq/xauthentication.hpp"
-#include "xeus-zmq/xmiddleware.hpp"
-#include "xeus-zmq/xzmq_serializer.hpp"
+#include "xeus/xkernel_configuration.hpp"
+#include "xeus-zmq/xzmq_context.hpp"
+#include "xipc_client.hpp"
 
 namespace nl = nlohmann;
 
 int main(int, char**)
 {
-    zmq::context_t context;
-    zmq::socket_t cli(context, zmq::socket_type::dealer);
-    xeus::xguid socket_id = xeus::new_xguid();
-    //cli.set(ZMQ_IDENTITY, socket_id.c_str(), socket_id.size());
-    cli.set(zmq::sockopt::linger, xeus::get_socket_linger());
-    cli.connect(xeus::get_end_point("ipc", "localhost", "shell"));
+    auto context_ptr = xeus::make_zmq_context();
+    xeus::xcontext& context = *context_ptr;
```

Comment:
I just realized the constructor for `xcontext` is defined in the protected block is xeus. So I guess this is the way we can use a xcontext reference ?

Reply:
Yes, `xcontext` is an abstract class used for type erasing the right type. No need to capture the reference though, you can directly dereference the pointer when you need it.